src/Database/Schema/Collection.php

```diff
@@ -158,7 +157,17 @@ protected function _reflect(string $stage, string $name, array $config, TableSch
             throw new DatabaseException($e->getMessage(), 500, $e);
         }
         foreach ($statement->fetchAll('assoc') as $row) {
-            $this->_dialect->{$convertMethod}($schema, $row);
+            $this->getDialect()->{$convertMethod}($schema, $row);
         }
     }
+
+    /**
+     * Setups the schema dialect to be used for this collection.
+     *
+     * @return \Cake\Database\Schema\SchemaDialect
+     */
+    private function getDialect(): SchemaDialect
```

Comment:
This could be protected right away, if we moved this into 5.next (5.2 to be released in the next month/weeks at some point anyway).